In [2]:
%run MLR_Functions.ipynb
import pandas as pd

In [3]:
df_rice_y = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_crop_yield.csv")
df_fert_use = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_fertilizer_usage.csv")
df_cropland = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_cropland.csv")
df_energy = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_energy_usage.csv")
df_pop = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_urban_rural_pop.csv")
df_water = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_water_usage.csv")
df_pesticide = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_pesticide_usage.csv")
df_pesticide_imports = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_pesticide_imports.csv")
df_air_dep = pd.read_csv("/Users/yxlsa/Desktop/SUTD/DTPIII/Cleaned Data/df_air_deposition_nitrogen.csv")
#print(df_pesticide_imports)

In [4]:
df_THAI = pd.DataFrame()
df_THAI["Yield"] = df_rice_y["Thailand Rice Yield"]
df_THAI["Cropland"] = df_cropland["Thailand Crop Land"]
df_THAI["Energy"] = df_energy["Thailand energy/cropland"]
df_THAI["Fert"] = df_fert_use["Thailand"]
df_THAI["Rural Pop"] = df_pop["thailand rural/total"]
df_THAI["Urban Pop"] = df_pop["thailand urban/total"]
df_THAI["Water"] = df_water["wf_tot_m3_t_thailand"]
df_THAI["Pesticide"] = df_pesticide["Thailand Pesticide"]
df_THAI["Pesticide Import"] = df_pesticide_imports["Thailand"]
df_THAI["Air Deposition of Nitrogen"] = df_air_dep["Thailand Air Deposition of Nitrogen"]

display(df_THAI)

Yield    Cropland    Energy        Fert  Rural Pop  Urban Pop  \
0   1.9556  20603.0000  0.016774   50.660000   0.723072   0.301452   
1   2.2534  20726.0000  0.016327   46.420002   0.718609   0.302042   
2   2.1745  20574.0000  0.020647   54.330000   0.713454   0.302332   
3   2.0497  20445.0000  0.022891   71.160000   0.707804   0.302387   
4   2.3521  20538.0000  0.016827   66.939995   0.702430   0.302546   
5   2.4158  20631.0000  0.017973   73.840000   0.698021   0.303106   
6   2.4098  20724.0000  0.021540   75.750000   0.694821   0.304188   
7   2.3788  20817.0000  0.028534   75.400000   0.692356   0.305585   
8   2.4180  20909.0000  0.036329   85.030000   0.690393   0.307211   
9   2.4244  21002.0000  0.027940   91.580000   0.688809   0.309013   
10  2.6128  21095.0000  0.026281   80.160000   0.684960   0.313318   
11  2.8739  21188.0000  0.030244   85.370000   0.673389   0.324936   
12  2.9338  21281.0000  0.033156   88.979996   0.661119   0.336555   
13  2.9339  21374.0000  0.037560  111.420000   0.648365   0.348208   
14  2.8895  21467.0000  0.037565   97.750000   0.635296   0.359974   
15  2.9974  21560.0000  0.040909   83.240005   0.622122   0.371863   
16  2.9503  21653.0000  0.039902   85.320000   0.608692   0.383839   
17  3.0441  21745.0000  0.044369   98.570000   0.594834   0.395722   
18  2.9974  21838.0000  0.046488   91.710010   0.580752   0.407626   
19  2.9079  21931.0000  0.052200   87.869995   0.566677   0.419583   
20  2.9921  22024.0000  0.054922  113.990005   0.552709   0.431740   
21  3.1867  22028.0000  0.049682  116.860000   0.543503   0.439284   
22  3.1865  22032.0000  0.062255  114.970000   0.535190   0.445823   
23  3.1463  22036.2654  0.057505  127.140000   0.526973   0.452395   
24  3.0586  22042.0000  0.067616  121.170000   0.518863   0.459066   
25  2.8506  22046.0000  0.063195  103.760000   0.510879   0.465836   
26  2.9740  22042.0000  0.043608  112.460000   0.502792   0.472514   
27  3.0690  22047.0000  0.048660  126.970000   0.494668   0.479088   
28  3.0380  22045.6965  0.059603  105.150000   0.486829   0.485831   
29  2.9164  22200.0000  0.075892   85.030000   0.479236   0.492694   

          Water  Pesticide  Pesticide Import  Air Deposition of Nitrogen  
0   2801.857667       0.91          1.503185                      6.8414  
1   2559.945281       0.91          1.229454                      6.6988  
2   2614.282257       0.92          1.272915                      6.9686  
3   2440.991593       0.92          1.230838                      7.2271  
4   2249.798274       1.00          1.463031                      7.4001  
5   2387.660821       1.17          1.567750                      7.6855  
6   2071.595781       1.14          2.057729                      7.9887  
7   2345.042479       1.15          2.046642                      7.7375  
8   2419.960932       0.92          1.556494                      8.7682  
9   2048.445773       1.08          2.444729                      8.2513  
10  1934.570175       0.96          2.498689                      8.2272  
11  1954.076511       1.45          2.626778                      7.9752  
12  1836.180279       1.92          3.297674                      7.9139  
13  1856.321877       2.40          3.416735                      7.9106  
14  1813.644449       3.06          4.650945                      8.9125  
15  1911.681194       2.03          3.648075                      9.4916  
16  1701.732209       1.90          4.706090                      8.1991  
17  1864.368467       3.07          5.625965                      8.9677  
18  1735.114610       3.00          5.335703                      8.3495  
19  1751.082915       3.10          5.771591                      8.7588  
20  1915.324183       3.13          5.887516                      9.0397  
21  1547.424761       3.96          7.752465                      8.6757  
22  1683.856891       3.17          7.415428                      9.0002  
23  1653.161377       0.37        

In [17]:
display(df_THAI.corr()[["Yield"]])


Yield
Yield                       1.000000
Cropland                    0.912625
Energy                      0.794211
Fert                        0.846633
Rural Pop                  -0.817261
Urban Pop                   0.800804
Water                      -0.941927
Pesticide                   0.584576
Pesticide Import            0.822295
Air Deposition of Nitrogen  0.828354

Because the number of data points we have is small, the number of test data points we have for our single variable linear regressions are too few to establish an r^2 value that represents the data well. (Independent Variable vs Rice Yield)

Additionally, many of the single variable linear regression results are inconsistent with the preliminary scatter plots that we made when we cleaned up the data.

Hence we opted to use correlation to determine whether linear relationships exist betweeen our independent variables and Rice Yields, and whether we should accept/reject these independent variables. 

For this we used a threshold of 0.7. i.e. We reject independent variables with correlation values less than 0.7.

In the case for Thailand, we will drop the Pesticide (Pesticide Usage) data, since 0.584576 < 0.7.